curriculum/challenges/english/07-scientific-computing-with-python/learn-encapsulation-by-building-a-projectile-trajectory-calculator/660400f02cf3a76795b82d4e.md

```diff
@@ -0,0 +1,73 @@
+---
+id: 660400f02cf3a76795b82d4e
+title: Step 4
+challengeType: 20
+dashedName: step-4
+---
+
+# --description--
+
+The first thing to set up is a method that calculates the displacement of the projectile, which is the horizontal space traveled from the throw to when it touches the ground.
+
+Create a function `__calculate_displacement`, which has only `self` as parameter, and return the displacement of the projectile.
+
+Use the following formula to compute the projectile displacement:
+\\[ d = \frac{v^2 \cdot \cos(\theta) \cdot \left(\sin(\theta) + \sqrt{\sin^2(\theta) + \frac{2 \cdot g \cdot h}{v^2}}\right)}{g} \\]
+
+In which \\(d\\) is the displacement, \\(v\\) is the starting speed, \\(\theta\\) is the angle and \\(h\\) is the starting height of the projectile.
+For \\(g\\) you can use the `GRAVITATIONAL_ACCELERATION` variable that has been added on line 3 of the code.
```

Comment:
We can ask the camper to use the `GRAVITATIONAL_ACCELERATION` as `9.81` instead of creating a line of code ourselves.

Reply:
I did not add this step, but having this as a value outside means that we can simulate for example a moon environment just changing this value.
I will add the step to have the camper add this.